internal/fleet/integration_policy/input_value.go

```diff
@@ -0,0 +1,376 @@
+	if len(defaultStreams) == 0 {
+		return streams, nil
+	}
+
+	// If streams is not known, create new streams from defaults
+	if !utils.IsKnown(streams) {
+		streamsMap := make(map[string]integrationPolicyInputStreamModel)
+		for streamID, streamDefaults := range defaultStreams {
+			streamsMap[streamID] = integrationPolicyInputStreamModel(streamDefaults)
+		}
+		return types.MapValueFrom(ctx, getInputStreamType(), streamsMap)
+	}
+
+	// Convert streams to model
+	var diags diag.Diagnostics
+	streamsMap := utils.MapTypeAs[integrationPolicyInputStreamModel](ctx, streams, path.Root("streams"), &diags)
+	if diags.HasError() {
+		return streams, diags
+	}
+
+	// Apply defaults to each stream
+	for streamID, streamDefaults := range defaultStreams {
+		stream, exists := streamsMap[streamID]
+		if !exists {
+			// Stream not configured, use defaults
+			streamsMap[streamID] = integrationPolicyInputStreamModel(streamDefaults)
+			continue
+		}
+
+		// Apply defaults to existing stream
+		if !utils.IsKnown(stream.Enabled) && utils.IsKnown(streamDefaults.Enabled) {
+			stream.Enabled = streamDefaults.Enabled
+		}
+		varsWithDefaults, d := applyDefaultsToVars(stream.Vars, streamDefaults.Vars)
+		diags.Append(d...)
+		if diags.HasError() {
+			return streams, diags
+		}
+		stream.Vars = varsWithDefaults
+		streamsMap[streamID] = stream
+	}
+
+	return types.MapValueFrom(ctx, getInputStreamType(), streamsMap)
+}
+
+// compareStreams compares two inputs' streams after defaults have been applied
+func compareStreams(ctx context.Context, oldInput, newInput integrationPolicyInputsModel) (bool, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	// Handle null/unknown cases
+	if oldInput.Streams.IsNull() && newInput.Streams.IsNull() {
+		return true, diags
+	}
+	if oldInput.Streams.IsUnknown() && newInput.Streams.IsUnknown() {
+		return true, diags
+	}
+	if oldInput.Streams.IsNull() != newInput.Streams.IsNull() || oldInput.Streams.IsUnknown() != newInput.Streams.IsUnknown() {
+		return false, diags
+	}
+
+	// Convert both maps to model
+	oldStreamsMap := utils.MapTypeAs[integrationPolicyInputStreamModel](ctx, oldInput.Streams, path.Root("streams"), &diags)
+	if diags.HasError() {
+		return false, diags
+	}
+
+	newStreamsMap := utils.MapTypeAs[integrationPolicyInputStreamModel](ctx, newInput.Streams, path.Root("streams"), &diags)
+	if diags.HasError() {
+		return false, diags
+	}
+
+	// Filter out disabled streams from both maps
+	enabledOldStreams := filterEnabledStreams(oldStreamsMap)
+	enabledNewStreams := filterEnabledStreams(newStreamsMap)
+
+	// Check if the number of enabled streams is the same
+	if len(enabledOldStreams) != len(enabledNewStreams) {
+		return false, diags
+	}
+
+	// Compare each enabled stream
+	for streamID, oldStream := range enabledOldStreams {
+		newStream, exists := enabledNewStreams[streamID]
+		if !exists {
+			return false, diags
+		}
+
+		// Compare enabled flags
+		if !oldStream.Enabled.Equal(newStream.Enabled) {
```

Comment:
why do we compare `enable` attributes if we are considering only enabled streams?